python/kvikio/tests/test_http_io.py

```diff
@@ -100,3 +147,40 @@ def test_no_range_support(http_server, tmpdir, xp):
             OverflowError, match="maybe the server doesn't support file ranges?"
         ):
             f.read(b, size=10, file_offset=10)
+
+
+def test_retry_http_500_ok(tmpdir, xp):
+    a = xp.arange(100, dtype="uint8")
+    a.tofile(tmpdir / "a")
+
+    error_counter = ErrorCounter()
+
+    with LocalHttpServer(
+        tmpdir,
+        max_lifetime=60,
+        handler=HTTP500Handler,
+        handler_options={"error_counter": error_counter},
+    ) as server:
```

Comment:
```suggestion
    with LocalHttpServer(
        tmpdir,
        max_lifetime=60,
        handler=HTTP500Handler,
        handler_options={"error_counter":  ErrorCounter()},
    ) as server:
```

Reply:
Or will this not work because `LocalHttpServer` clear or deep copy the `handler_options` dict?